### PR TITLE
Feature/24 09 2022/init files table

### PR DESCRIPTION
**Pull request by tungjj**

### CODE REVIEW

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@RestController
+@RequestMapping(value = "file-controller")
+public class FileController extends AbstractController<FileService> {
+    @PostMapping(value = "create-file")
+    @SecurityRequirement(name = "Bearer Authentication")
+    void createFile(@RequestBody FileRequest fileRequest) {
+        service.createFile(fileRequest);
+    }
+
+    @GetMapping(value = "get-files-by-user-id")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/controller/FileController.java:28]: change to /get-file for single file, get files for multiple and add prefix such as /user, /public, /admin

**Review by ericchentch — CHANGES_REQUESTED** [review]
check review comments, rebase to get new update in abstract mongo repo

File: src/main/java/edunhnil/project/forum/api/config/SecurityConfig.java [hunk 2]
```diff
@@ -32,7 +32,8 @@ public void configure(HttpSecurity http) throws Exception {
                                                 "/account/user/getProfile",
                                                 "/post/user/getListPost",
                                                 "/post/user/addNewPost", "/comment/user/addNewComment/**",
-                                                "/like/user/**")
+                                                "/like/user/**",
+                                                "/files-controller/user/**")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/config/SecurityConfig.java:36]: change to /files not files-controller

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import edunhnil.project.forum.api.dto.commonDTO.CommonResponse;
+import edunhnil.project.forum.api.dto.commonDTO.ListWrapperResponse;
+import edunhnil.project.forum.api.dto.fileDTO.FileRequest;
+import edunhnil.project.forum.api.dto.fileDTO.FileResponse;
+import edunhnil.project.forum.api.service.fileService.FileService;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+
+@RestController
+@RequestMapping(value = "files-controller")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/controller/FileController.java:20]: change to /files

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+public class FileController extends AbstractController<FileService> {
+    @PostMapping(value = "user/create-file")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/controller/FileController.java:22]: user/create

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    @SecurityRequirement(name = "Bearer Authentication")
+    void createFile(@RequestBody FileRequest fileRequest) {
+        service.createFile(fileRequest);
+    }
+
+    @GetMapping(value = "user/get-files-by-user-id")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/controller/FileController.java:28]: user/get-files

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    @SecurityRequirement(name = "Bearer Authentication")
+    ResponseEntity<CommonResponse<ListWrapperResponse<FileResponse>>> getFilesByUserId(String userId) {
+        return response(service.getFilesByUserId(userId), "Success");
+    }
+    
+    @GetMapping(value = "user/get-file-by-id")
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/controller/FileController.java:34]: user/get-file

File: src/main/java/edunhnil/project/forum/api/dto/fileDTO/FileEnum.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package edunhnil.project.forum.api.dto.fileDTO;
+
+public enum FileEnum {
```
Comment (ericchentch) [inline src/main/java/edunhnil/project/forum/api/dto/fileDTO/FileEnum.java:3]: discard this file

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java (hunk 1 above)
Comment (tungjj) [reply src/main/java/edunhnil/project/forum/api/controller/FileController.java]: done

File: src/main/java/edunhnil/project/forum/api/config/SecurityConfig.java (hunk 2 above)
Comment (tungjj) [reply src/main/java/edunhnil/project/forum/api/config/SecurityConfig.java]: done

File: src/main/java/edunhnil/project/forum/api/controller/FileController.java (hunk 3 above)
Comment (tungjj) [reply src/main/java/edunhnil/project/forum/api/controller/FileController.java]: done

**Comment by tungjj** [comment]
Need to review

**Review by ericchentch — APPROVED** [review]
LG

**Pull request merged**